CPP_STYLE_GUIDE.md

```diff
@@ -200,6 +201,42 @@ void FooConsumer(std::unique_ptr<Foo> ptr);
 
 Never use `std::auto_ptr`. Instead, use `std::unique_ptr`.
 
+### Avoid non-const references
+
+Using non-const references often obscures which values are changed by an
```

Comment:
### Avoid non-const reference arguments

The idiomatic semantic for non-const reference argument is that of an [in/out argument](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inout). It is advised to use const pointer instead for in/out parameter, as per the [Google style guide](https://google.github.io/styleguide/cppguide.html#Reference_Arguments).
Other uses of non-const reference should only when they don't interfere with readability.

Reply:
Suggestion.

Reply:
... and when the semantics of the API in which they appear are unambiguous.

Reply:
> The idiomatic semantic for non-const reference argument is that of an [in/out argument](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inout).

That sounds like it’s the main use – that’s not the case. Also, “idiomatic” may be confusing, because it is *not* idiomatic for our or V8’s codebase, which make up the vast majority of C++ code here.

> It is advised to use a pointer instead for in/out parameter, as per the [Google style guide](https://google.github.io/styleguide/cppguide.html#Reference_Arguments).

I think this is implied by the current text and the examples. Do you feel strongly about linking to external documents?

Reply:
I do think that linking to external document is good practice. Since both the Google SG, and the Cpp Core Guidelines give a broader context. Also AFAIU we base our style guide on google's.
I'll try to rephrase my suggestion above.